ansible/rke2/airgap/roles/rke2_upgrade/tasks/main.yml

```diff
@@ -0,0 +1,409 @@
+---
+- name: RKE2 Upgrade Tasks
+  block:
+    - name: Check current RKE2 version
+      shell: |
+        if [ -f /usr/local/bin/rke2 ]; then
+          /usr/local/bin/rke2 --version | head -1 | awk '{print $3}'
+        else
+          echo "not-installed"
+        fi
+      register: current_rke2_version
+      changed_when: false
+      become: true
+
+    - name: Display current and target versions
+      debug:
+        msg: |
+          Current RKE2 Version: {{ current_rke2_version.stdout }}
+          Target RKE2 Version: {{ rke2_version }}
+          Upgrade needed: {{ current_rke2_version.stdout != rke2_version }}
+
+    - name: Skip upgrade if versions match
+      debug:
+        msg: "RKE2 is already at version {{ rke2_version }}. Skipping upgrade."
+      when: current_rke2_version.stdout == rke2_version
```

Comment:
This could be confusing because we run the rke2-upgrade task, expecting it to do something, and instead it just skips. Maybe we need it to error out in the case where the version matches already?